ops/model.py

```diff
@@ -985,6 +1006,52 @@ def location(self):
         return self._location
 
 
+class Container:
+    """Represents a named (Kubernetes) container in an application.
+
+    Attributes:
+        name: The name of the container from metadata.yaml (eg, 'postgres').
+    """
+
+    def __init__(self, name):
+        self.name = name
+
+        socket_path = '/charm/containers/{}/pebble/.pebble.socket'.format(name)
+        self._pebble = pebble.API(socket_path=socket_path)
+
+    @property
+    def pebble(self) -> 'pebble.API':
+        """Return the low-level Pebble API instance for this container."""
+        return self._pebble
+
+    def autostart(self):
+        """Autostart all default services."""
+        self._pebble.autostart_services()
+
+    def start(self, *service_names: typing.List[str]):
+        """Start given service(s) by name."""
+        self._pebble.start_services(service_names)
+
+    def stop(self, *service_names: typing.List[str]):
+        """Stop given service(s) by name."""
+        self._pebble.stop_services(service_names)
+
+    def add_layer(self, config: typing.Union[str, typing.Dict, 'pebble.Config']):
+        """Dynamically add a service (Pebble) configuration layer.
+
+        Args:
+            config: A YAML string, config dict, or pebble.Config object
+                containing the Pebble configuration layer to add
+        """
+        if isinstance(config, (str, dict)):
+            config = pebble.Config(config)
+        self._pebble.add_layer(config)
```

Comment:
Shouldn't we go the opposite way: if we have something that is not a string, transform it into a string. That enables people to use configurations that are known to Pebble but not to the framework yet, as otherwise it'd fail to parse. It also avoids the unnecessary decode+encode roundtrip.

Reply:
Good call -- done.